draft-ietf-anima-brski-prm.md

```diff
@@ -1298,27 +1342,29 @@ The header of the RVR SHALL contain the following parameter as defined for JWS {
 
 The payload of the RVR MUST contain the following parameter as part of the voucher-request as defined in {{!RFC8995}}:
 
-* created-on: current date and time in yang:date-and-time format of RVR creation
+* `created-on`: SHALL contain the current date and time at RVR creation as standard date/time string as defined in {{Section 5.6 of !RFC3339}};
+
+* idevid-issuer: issuer value from the pledge IDevID certificate
 
 * nonce: copied from the PVR
 
 * serial-number: product-serial-number of pledge.
-  The registrar MUST verify that the IDevID certificate subject serialNumber of the pledge (X520SerialNumber) matches the serial-number value in the PVR.
+  The registrar MUST verify that the X520SerialNumber field of the pledge IDevID certificate matches the serial-number value in the PVR.
   In addition, it MUST be equal to the serial-number value contained in the agent-signed data of PVR.
 
 * assertion: voucher assertion requested by the pledge (agent-proximity).
```

Comment:
Is this only included if the pledge included it?
Is the pledge allowed to not include it? Overall, this is unclear:

* `assertion`: SHALL contain the requested voucher assertion "agent-proximity" <-- SHALL, i.e., mandatory, but "requested" -- only if requested? Or rather "chosen" type of (mandatory) assertion?
* "The MASA MAY verify..." <-- MASA MAY irgnore, i.e., optional
* "If the assertion "agent-proximity" is requested" <-- sounds optional
* "+-- assertion?" (RFC8995) <-- optional field
* "This MUST be populated in a pledge's voucher-request when a proximity assertion is requested." (RFC8995) <-- sounds very optional

What is the role of the registrar for providing this information in the next line? It just MUST forward information or it MUST insert information if missing to assure?

Reply:
The pledge includes the agent-proximity assertion as "request". "requested voucher assertion" relates to the PVR.
To be able to provide this assertion in the voucher, the MASA MUST verify agent-signed data. Hence, the sentence should be:

agent-signed-data: The MASA MUST verify this data to issue "agent-proximity" assertion. 

Good finding!

Reply:
The pledge MUST include or MAY include the request for agent-proximity assertion?
It the moment it sounds like it is optional, i.e., the pledge may just omit the "assertion" field unless the pledge explicitly has a request for this assertion.

Reply:
In the context of BRSKI-PRM it MUST include it as this is the highest assertion it can request. As the intention is to not require the configuration of a pledge upfront, it would be difficult to determine if or if not the assertion is to be included.

Reply:
Done